assets/react/v3/entries/course-builder/components/layouts/Notebook.tsx

```diff
@@ -0,0 +1,366 @@
+
+			const newX = Math.min(Math.max(event.clientX - offset.x, 0), windowWidth - notebookWidth);
+			const newY = Math.min(Math.max(event.clientY - offset.y, 0), windowHeight - notebookHeight);
+
+			warpper.style.left = `${newX}px`;
+			warpper.style.top = `${newY}px`;
+		}, 10);
+
+		const handleMouseUp = () => {
+			setIsDragging(false);
+
+			if (isDefined(wrapperRef.current)) {
+				const wrapper = wrapperRef.current;
+				const { left, top, height, width } = wrapper.style;
+				const currentNotebookData: NotebookData = JSON.parse(getFromLocalStorage(LocalStorageKeys.notebook) || '{}');
+				if (left === 'auto' || top === 'auto') return;
+				setToLocalStorage(
+					LocalStorageKeys.notebook,
+					JSON.stringify({
+						...currentNotebookData,
+						position: {
+							left,
+							top,
+							height,
+							width,
+						},
+					})
+				);
+			}
+		};
+
+		document.addEventListener('mousemove', handleMouseMove);
+		document.addEventListener('mouseup', handleMouseUp);
+
+		return () => {
+			document.removeEventListener('mousemove', handleMouseMove);
+			document.removeEventListener('mouseup', handleMouseUp);
+		};
+	}, [isDragging, offset]);
+
+	// biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
+	useEffect(() => {
+		if (isResizing && isDefined(wrapperRef.current)) {
+			const wrapper = wrapperRef.current;
+			const { left, top, height, width } = wrapper.style;
+			const notebookData: NotebookData = JSON.parse(getFromLocalStorage(LocalStorageKeys.notebook) || '{}');
+			setToLocalStorage(
+				LocalStorageKeys.notebook,
+				JSON.stringify({
+					...notebookData,
+					position: {
+						left,
+						top,
+						height,
+						width,
+					},
+				})
+			);
+		}
+	}, [isResizing, size]);
+
+	useEffect(() => {
+		if (isCollapsed && isDefined(wrapperRef.current) && !isFloating) {
+			setIsFloating(false);
+			const wrapper = wrapperRef.current;
+
+			wrapper.style.left = 'auto';
+			wrapper.style.top = 'auto';
+			wrapper.style.width = '360px';
+		}
+
+		if (!isCollapsed && isFloating && isDefined(wrapperRef.current)) {
+			const notebookData: NotebookData = JSON.parse(getFromLocalStorage(LocalStorageKeys.notebook) || '{}');
+			const { left, top, height, width } = notebookData.position || {
+				left: 'auto',
+				top: 'auto',
+			};
+
+			if (left && top) {
+				const wrapper = wrapperRef.current;
+
+				wrapper.style.left = left === 'auto' ? `${window.innerWidth - notebook.MIN_NOTEBOOK_WIDTH}px` : left;
+				wrapper.style.top = top === 'auto' ? `${window.innerHeight - notebook.MIN_NOTEBOOK_HEIGHT}px` : top;
+				if (isDefined(height)) {
+					wrapper.style.height = height;
+				}
+				if (isDefined(width)) {
+					wrapper.style.width = width;
+				}
+			}
+		}
+	}, [isCollapsed, isFloating]);
+
+	useEffect(() => {
+		const notebookData: NotebookData = JSON.parse(getFromLocalStorage(LocalStorageKeys.notebook) || '{}');
+		if (notebookData.notes) {
```

Comment:
Use a newline before and after any block.
```suggestion

		if (notebookData.notes) {
```